verification/verify_test.go

```diff
@@ -57,19 +61,42 @@ func TestVerify(t *testing.T) {
 	assert.Nil(res)
 	assert.Equal(ErrPixelsAbsent, err)
 
-	// check pixel count fails
+	// check pixel count fails w/ external verifier
 	data.Segments = append(data.Segments, &net.TranscodedSegmentData{Url: "def", Pixels: verifier.results.Pixels[1]})
 	assert.Len(data.Segments, len(verifier.results.Pixels)) // sanity check
 	res, err = sv.Verify(&Params{Results: data})
 	assert.Nil(res)
 	assert.Equal(ErrPixelMismatch, err)
 
-	// Check pixel count succeeds
+	// Check pixel count succeeds w/ external verifier
 	data.Segments[0].Pixels = verifier.results.Pixels[0]
 	res, err = sv.Verify(&Params{Results: data})
 	assert.Nil(err)
 	assert.NotNil(res)
 
+	// check pixel count fails w/o external verifier
+	sv = NewSegmentVerifier(&Policy{Verifier: nil, Retries: 2}) // reset
+	data = &net.TranscodeData{Segments: []*net.TranscodedSegmentData{
+		{Url: "abc", Pixels: verifier.results.Pixels[0] + 1},
+	}}
+	data.Segments = append(data.Segments, &net.TranscodedSegmentData{Url: "def", Pixels: verifier.results.Pixels[1]})
+	assert.Len(data.Segments, len(verifier.results.Pixels)) // sanity check
+	res, err = sv.Verify(&Params{Results: data, Orchestrator: &net.OrchestratorInfo{}})
+	assert.Nil(res)
+	assert.Equal(ErrPixelsAbsent, err)
```

Comment:
Can we also add a test checking for `ErrPixelsMismatch` in the case where `Params.Renditions` is populated and the slice is the same length as the length of `Segments` except the actual pixel counts are incorrect?

Reply:
added in 381cf5938ce6f66d5a2af293a461282c1d0bff49